seonoh/ArchitectureStudy/app/src/main/java/com/example/seonoh/seonohapp/CoinFragment.kt

```diff
@@ -1,33 +1,31 @@
 package com.example.seonoh.seonohapp
 
 import android.os.Bundle
-import android.util.Log
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import androidx.fragment.app.Fragment
 import com.example.seonoh.seonohapp.model.CurrentPriceInfoModel
 import com.example.seonoh.seonohapp.model.UseCoinModel
 import com.example.seonoh.seonohapp.network.CoinRequest
-import com.example.seonoh.seonohapp.network.RetrofitCreator
 import com.example.seonoh.seonohapp.util.CalculateUtils
 import kotlinx.android.synthetic.main.coin_fragment.*
-import kotlinx.android.synthetic.main.coin_fragment.view.*
 
 
 class CoinFragment : Fragment(), CoinRequest.BaseResult<ArrayList<CurrentPriceInfoModel>> {
 
     lateinit var mAdapter: CoinAdapter
     private var marketName: String? = null
+    private val coinRepository by lazy { CoinRepositoryImpl() }
+
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
 
         marketName = arguments?.getString(MARKET)
-
-        requestData()
+        coinRepository.sendCurrentPriceInfo(this,marketName!!)
```

Comment:
marketName 이 null 인 경우는 없을까요?

Reply:
네 반영했습니다. 감사합니다 !!

0bd4e37

Reply:
arguments 가 null 인 경우, marketName을 nonnull 로 casting 하면 NPE 발생할 여지가 있어서
sendCurrentPriceInfo 에서 두 번째 파라미터 타입을 nullable 타입으로 하는 것을 의도로 하였습니다.

Reply:
아!! 네 감사합니다 !

b9b5169

Reply:
```suggestion
        marketName = arguments?.getString(MARKET)
        marketName?.let {
            coinRepository.sendCurrentPriceInfo(
                this,
                it
            )
        } ?: Toast.makeText(
            activity,
            "마켓데이터를 불러올 수 없습니다.",
            Toast.LENGTH_LONG
        ).show()        
```
이런 식으로 하시면 될 것 같습니다.

Reply:
감사합니다 ! 
1c67a4a